src/impl_methods.rs

```diff
@@ -2068,7 +2068,12 @@ where
         let view_stride = self.strides.axis(axis);
         // use the 0th subview as a map to each 1d array view extended from
         // the 0th element.
-        self.index_axis(axis, 0).map(|first_elt| {
+        let first_subview = if view_len != 0 {
+            self.index_axis(axis, 0)
+        } else {
+            ArrayView::<A, D::Smaller>::new_empty()
```

Comment:
@jturner314 correct me if I wrong, but array with shape `&[3, 0, 4]` may not contain any elements since the total number of elements is `3 * 0 * 4 = 0`. Example:
```rust
let a = Array3::from_elem((3, 0, 4), 1.0);
assert_eq!(a.is_empty(), true);
```
But array of shape `&[3, 4]` should contain 12 elements. Where should we take 12 more elements from in this case?

Reply:
Thanks, I got the idea